public/docs/ts/latest/guide/component-communication.jade

```diff
@@ -0,0 +1,707 @@
+  
+figure.image-display
+  img(src="/resources/images/devguide/component-communication/parent-and-children.png" alt="Input data binding")
+  
+:marked
+  When we run the app and click the **Invite heroes** button, the application UI reflects this stucture:
+
+figure.image-display
+  img(src="/resources/images/devguide/component-communication/heroes-invited-ui.png" alt="Heroes are invited")
+  
+:marked
+  ### Using `@Input`
+  
+  `Input` is defined in the core module of Angular, and we need to import it so that we can use it:
+  
++makeExample('component-communication/ts/app/invite-heroes/hero-panel.ts', 'import', null, {blk: /(Input)/g})
+
+:marked
+  `@Input()` always must be used with parentheses:
+
++makeExample('component-communication/ts/app/invite-heroes/hero-panel.ts', 'input-def', null, {pnk: /(@Input\(\))/g})
+
+:marked
+.alert.is-critical
+  :marked
+    It's easy to forget the parentheses. Our application will fail mysteriously if we do.
+:marked
+  Instead of using the `@Input()` annotation, we can use the `inputs` annotation property of `@Component()` to specify input properties.
+
+  Using the alternative method, we could have defined the `hero` input property this way:
+  
++makeExample('component-communication/ts/fragments/using-inputs.ts', null, null, {pnk: /(inputs)/g})
+
+:marked
+.alert.is-helpful
+  :marked
+    In this chapter, we are age going to use the `@Input()` annotation.
+:marked
+  ### The convenience of input data binding
+  
+  The change detection mechanism of Angular makes input data binding easy-to-use. We can utilize the input property values in expressions, too,
+  and it makes displaying data simple.
+  
+  We have already implemented the functionality behind the __Invite heroes__ function with input data binding. The same mechanism allows us
+  to implement announcing jobs. As the altered definition of `HeroPanel` shows, now it has another input property, `request`, which has two
+  occurrences within the template:
+
++makeExample('component-communication/ts/app/send-job-request/hero-panel.ts', 'component', 'send-job-request/hero-panel.ts (extract)', {pnk: /(&quot;request&quot;|{{request \|\|.*\}\})/g})
+
+:marked
+  As the value of `request` changes, the value of the `class` attribute is updated, as well as the text of the `<h4>` tag through the 
+  `request || 'No job announced'` expression. The `HeroJobBoard` side of the input data binding is as simple as we expect:
+  
++makeExample('component-communication/ts/fragments/sending-request.ts', null, 'send-job-request/hero-job-board.ts (extract)', {pnk: /(\[request\]=request)/g})
+
+:marked
+  After inviting heroes, clicking the **Ask** button invokes `announceJob()` that sets the `request` property of `HeroJobBoard` &mdash; and Angular propagates this
+  change to all `HeroPanel` instances:
+
+figure.image-display
+  img(src="/resources/images/devguide/component-communication/send-job-request-ui.png" alt="Job request is sent")
+  
```

Comment:
Missed an important opportunity to talk about how the child can know when an input changes.

Suppose we bound to a single component ... a message component ...
```
<message-box [message]="currentMessage"></message-box>
```
How does the `MessageBox` know when the `currentMessage` changes?

That's a really important question.

I can think of 2 ways (I learned them from Victor):
1. implement the `MessagBox.message` as a setter-only property.  The setter method is called when the message changes which is the opportunity for the message box to respond (e.g., by changing the color of the display based on whether the message is a success, warning, or error message).

1. Implement `ngOnChanges`. Angular calls this method when any input value (e.g., the message) changes. 

The first option is usually the simplest. The second option makes more sense when you want to respond to multiple input binding changes and, perhaps, coordinate among them. For example, if our component is bound to departure and arrival date properties, we'd want to know both dates before updating the display (e.g., showing an error if the user would arrive before departure).